core/templates/dev/head/pages/exploration_editor/translation_tab/AudioTranslationBarDirective.ts

```diff
@@ -362,20 +399,13 @@ oppia.directive('audioTranslationBar', [
           $scope.initAudioBar = function() {
             // This stops angular-recorder when user navigate while recording.
             if ($scope.recorder) {
-              if ($scope.recorder.status.isPlaying) {
-                $scope.recorder.playbackPause();
-              }
-              if ($scope.recorder.status.isRecording &&
+              if ($scope.isRecording &&
                 $scope.showRecorderWarning) {
                 $scope.recorder.stopRecord();
-                $rootScope.loadingMessage = 'loading';
-              }
-              if ($scope.recorder.status.isConverting) {
-                $rootScope.loadingMessage = 'loading';
-              }
-              if (recorderService && recorderService.getHandler()) {
-                recorderService.getHandler().stop();
-                recorderService.getHandler().clear();
+                cancelTimer();
+                // when empty it runs fine but when set to anything else
+                // stuck on loading screen animation
+                $rootScope.loadingMessage = '';
```

Comment:
So we populate "$rootScope.loadingMessage" when we are handling data or making ajax request and we don't want users to interact with the page until it's fully loaded, so once the ajax call is completed and page is rendered we use to set "$rootScope.loadingMessage = ''".

If you want to achieve any such behavior you can load "$rootScope.loadingMessage" with some text and once you want users to interact with your page you can make "$rootScope.loadingMessage = ''".

Reply:
I see I didn't know that I was basing this on the previous behavior and had no context of that. The previous behavior was https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/pages/exploration_editor/translation_tab/AudioTranslationBarDirective.ts#L368 the following. I am trying to keep minimal changes on the UI side.

Reply:
I look at an implementation of what you told me in [TranslationTabDirective.ts](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationTabDirective.ts#L50) . I don't see how this benefits from it UX wise. I think using modals and popovers as indicators might be questionable UX wise.

Reply:
Not needed so done.